src/test/java/org/opensearch/security/auth/UserInjectorTest.java

```diff
@@ -64,6 +64,47 @@ public void testValidInjectUser() {
         assertEquals(injectedUser.getRoles(), roles);
     }
 
+    @Test
+    public void testValidInjectUserIpV6() {
+        HashSet<String> roles = new HashSet<>();
+        roles.addAll(Arrays.asList("role1", "role2"));
+        threadContext.putTransient(
+            ConfigConstants.OPENDISTRO_SECURITY_INJECTED_USER,
+            "user|role1,role2|2001:db8:3333:4444:5555:6666:7777:8888:9200"
+        );
+        UserInjector.InjectedUser injectedUser = userInjector.getInjectedUser();
+        assertEquals("user", injectedUser.getName());
+        assertEquals(9200, injectedUser.getTransportAddress().getPort());
+        assertEquals("2001:db8:3333:4444:5555:6666:7777:8888", injectedUser.getTransportAddress().getAddress());
```

Comment:
can you add test-cases for shortened ipv6 format- https://en.wikipedia.org/wiki/IPv6_address#Representation

Reply:
Added one here: [a680e92](https://github.com/opensearch-project/security/pull/4399/commits/a680e921e3e5f866a5f91a05883bf547309e36ee). I tried to use an address that would cover the two scenarios I can think of: a sequence of 0's shortened to `::`, as well as leading 0's cut off (`0001` shortened to `1`).